### PR TITLE
Sparrow dom/facebook native share

### PR DESCRIPTION
**Pull request by sparrowDom**
### Description:

- Add facebook native share capabilities to Android / IOS
  - do so by intercepting WebView's navigation to Facebook's share dialog, extract the link information and trigger native FB sdk share
  - fix deep links to Facebook OriginProtocol page, since they are different on IOS/Android

**Important** to include cocoaPods dependacies open project using `OriginCatcher.xcworkspace` file instead of `OriginCatcher.xcodeproj`

### Checklist:

- [x] Test your work and double-check to confirm that you didn't break anything
- [ ] [Wrap any new text/strings with `fbt`](https://github.com/OriginProtocol/origin/tree/master/dapps/marketplace/translation#wrapping-text) for translation
- [ ] If there are any new/changed strings to translate, [`npm run translate`](https://github.com/OriginProtocol/origin/tree/master/dapps/marketplace/translation#integrating-translations)
- [ ] Update any relevant READMEs and [docs](https://github.com/OriginProtocol/origin/tree/master/docs)

### CODE REVIEW

**Comment by sparrowDom** [comment]
@tomlinton it is not really a good approach to include dependant libraries. I am thinking of removing them and letting the developer install cocoapods and fetch libraries using `pod install`. 

It is cleaner, but extra step for a user to make... what are your thoughts on that?

**Comment by tomlinton** [comment]
Yea I agree, keep it out if we can. We can just add it to the docs. It might be possible to do it automatically with a postinstall step in npm but we can look at that later.

**Comment by sparrowDom** [comment]
@tomlinton this is ready for another pass

File: mobile/README.md [hunk 1]
```diff
@@ -23,6 +23,10 @@ This guide will help you to install a local build of Origin Wallet on a device o
 
 - Xcode Command Line Tools: `xcode-select --install`
 
+- Install CocoaPods: `sudo gem install cocoapods`
+
+- (!) For the project to load with cocoa pods dependancies be sure to open the `OriginCatcher.xcworkspace` file instead of `OriginCatcher.xcodeproj`
```
Comment (tomlinton) [inline mobile/README.md]: Spelling, dependancies should be dependencies

File: mobile/src/screens/marketplace.js [hunk 2]
```diff
@@ -612,21 +628,88 @@ class MarketplaceScreen extends Component {
...
+        } else {
+          console.log(`Share success with postId: ${shareResult.postId}`)
+        }
+      }
+
+      /* After Facebook shows up the share dialog in dapp's WebView and user is not logged
+       * in it will redirect to login page. For that reason we return to the last dapp's
+       * url instead of triggering back.
+       */
+      if (shareHasBeenTriggeredRecenty) {
```
Comment (tomlinton) [inline mobile/src/screens/marketplace.js]: Typo/spelling

File: mobile/README.md (hunk 1 above)
Comment (sparrowDom) [reply mobile/README.md]: yup tnx 👍

File: mobile/src/screens/marketplace.js (hunk 2 above)
Comment (sparrowDom) [reply mobile/src/screens/marketplace.js]: tnx fixed

**Pull request merged**